registry/proxy/proxyregistry.go

```diff
@@ -193,8 +193,9 @@ func (r *remoteAuthChallenger) tryEstablishChallenges(ctx context.Context) error
 	r.Lock()
 	defer r.Unlock()
 
-	remoteURL := r.remoteURL + "/v2/"
-	challenges, err := r.cm.GetChallenges(remoteURL)
+	remoteURL := r.remoteURL
+	remoteURL.Path = "/v2"
```

Comment:
Not sure we have any tests for this or we explicitly support it, but not sure where else we are enforcing that the path is always "/v2" instead of something like "/api/registry/v2/". Also it was previously adding the trailing slash, I think we should keep that.